app/controllers/master/api/providers_controller.rb

```diff
@@ -157,6 +161,15 @@ def provider_account
     @provider_account ||= current_account.providers.without_deleted(!action_includes_deleted_providers?).find(params[:id])
   end
 
+  def signup_result_with_nil_token
+    signup_result = Signup::ResultWithAccessToken.new(account: provider_account, user: provider_account.admin_users.first)
+
+    # Signup::ResultWithAccessToken because it is the representer, but after it initializes, it builds an access token,
+    # which is a known design problem). So this access token needs to be set to nil before responding because it is an unsaved one
+    signup_result.access_token = nil
+    signup_result
+  end
```

Comment:
The reason is explained in the comment :)

Basically, the representer for this controller is `::Signup::ResultWithAccessToken`, but in all actions except `:create` the `access_token` is not the real one.

I wanted to avoid refactoring this specific and old code now for the feature that is not even public, so just added this method to remove duplication.

Reply:
OK, what confused me is the Signup term, which makes sense for `:create` but not for other actions. Definitely this needs a refactor or at least a renaming. But it doesn't need to be in this PR.